### ISSUE TITLE
HTTP 500, when opening any commits from a certain repository

### ISSUE THREAD

**Issue by keks24** [go-gitea/gitea#12825]
- Gitea version (or commit ref):
```bash
$ gitea --version
Gitea version 1.12.4 built with GNU Make 4.1, go1.14.8 : bindata, sqlite, sqlite_unlock_notify
```

- Git version:
```bash
$ git --version
git version 2.20.1
```

- Operating system:
```bash
$ uname --all
Linux gitea 5.4.60-1-pve #1 SMP PVE 5.4.60-2 (Fri, 04 Sep 2020 10:24:50 +0200) x86_64 GNU/Linux
```
```bash
$ < /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite

- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant

- Log gist:
[gitea.log](https://github.com/go-gitea/gitea/files/5214385/gitea.log)

## Description

I am getting an `HTTP 500` error, when I am trying to open any commit from my internal repository `l4d2-configuration`. The log file returns:

`PANIC:: runtime error: slice bounds out of range`

This also applies to a mirror repository on [gitea.com](https://gitea.com/keks24/l4d2-configuration/commit/69242277082f1a308bb5006ff1c00b2928b62dad).

For some reason it works on [try.gitea.io](https://try.gitea.io/keks24/l4d2-configuration/commit/69242277082f1a308bb5006ff1c00b2928b62dad) and on [codeberg.org](https://codeberg.org/keks24/l4d2-configuration/commit/69242277082f1a308bb5006ff1c00b2928b62dad); maybe because of some (custom) patches.

On [github.com](https://github.com/keks24/l4d2-configuration/commit/69242277082f1a308bb5006ff1c00b2928b62dad) and [gitlab.com](https://gitlab.com/keks24/l4d2-configuration/-/commit/69242277082f1a308bb5006ff1c00b2928b62dad) the commit can be displayed as well, but these platforms cannot be used as reference actually.

I can reproduce this issue, when I recreate the repository on my internal server and do a `git push --force`.

## Screenshots

![20200913-171701_screenshot](https://user-images.githubusercontent.com/13721064/93022174-f4b56300-f5e7-11ea-8c98-4583fa36c21f.png)

**Comment by keks24** [comment]
This also relates to #12823

**Comment by zeripath** [comment]
Please do not open duplicate bug reports. Keeping on top of Gitea's issues is hard enough.
This is not only a duplicate of #12823 but is a duplicate of #12816 which was already fixed in master by the time it was reported.

**Comment by zeripath** [comment]
Fixed by #12771 and its backport #12776
